### PR TITLE
Add integration tests for Ocr data validator endpoint

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-683

### Change description ###
Added integrations tests for ocr validation endpoint.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                .content(content))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.status").value("ERRORS"))
+            .andExpect(jsonPath("$.errors", hasSize(1)))
+            .andExpect(jsonPath("$.warnings", hasSize(0)))
+            .andExpect(jsonPath("$.errors[0]").value("last_name is missing"));
+    }
+
+    @Test
+    void should_return_warnings_when_ocr_form_data_has_missing_fields() throws Throwable {
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationTest.java]: missing_optional_fields ?

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                .content(content))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.status").value("WARNINGS"))
+            .andExpect(jsonPath("$.errors", hasSize(0)))
+            .andExpect(jsonPath("$.warnings", hasSize(1)))
+            .andExpect(jsonPath("$.warnings[0]").value("date_of_birth is missing"));
+    }
+
+    @Test
+    void should_return_401_when_service_auth_header_is_missing() throws Throwable {
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationTest.java]: need to add one more where service is not allowed to access the endpoint ?

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationTest.java (hunk 1 above)
Comment (aliveni) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationTest.java]: renamed the test

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationTest.java (hunk 2 above)
Comment (aliveni) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrValidationTest.java]: As discussed, this test scenario will not add any value as we have to mock the `AuthService` and it's already covered in unit test

**Pull request merged**